dhis-2/dhis-api/src/main/java/org/hisp/dhis/cache/LocalCache.java

```diff
@@ -102,14 +106,24 @@ public LocalCache( final CacheBuilder<V> cacheBuilder )
         {
             throw new IllegalArgumentException( "MappingFunction cannot be null" );
         }
-        return Optional
-            .ofNullable( Optional.ofNullable( caffeineCache.get( key, mappingFunction ) ).orElse( defaultValue ) );
+
+        V value = cache2kInstance.get( key );
+        if ( value == null )
+        {
+            value = mappingFunction.apply( key );
+        }
+
+        if ( value != null )
```

Comment:
Should we also cache null values? I think some methods have null as a valid response. The downside of not caching null in the scenario of methods which are computationally very expensive and where null is a valid output, leading to the same computation being done over and over unnecessarily.